Added configuration files and developed Maven sub-modules

NassimBtk opened the pull request:
Added configuration files and developed Maven sub-modules

Developed Maven modules:
* my-menu-tsps-core
* my-menu-tsps-web
* my-menu-tsps-lib
* my-menu-tsps-rest-api
* my-menu-tsps-plugin

Created docs folder

Created .github folder for workflow CI.

Updated .gitignore file

Update README.md file

NassimBtk committed:
Added configuration files and developed Maven sub-modules

Developed Maven modules:
* my-menu-tsps-core
* my-menu-tsps-web
* my-menu-tsps-lib
* my-menu-tsps-rest-api
* my-menu-tsps-plugin

Created docs folder

Created .github folder for workflow CI.

Updated .gitignore file

Update README.md file

bertysentry commented on my-menu-tsps-core/src/main/java/com/bmc/truesight/platform/etl/component/management/types/MyMenuComponent.java:
```diff
@@ -0,0 +1,323 @@
+	 * @return The username configured in this Component, or "admin" if null or
+	 *         empty
+	 */
+	private String getUsername() {
```
I guess this method is not required. I haven't seen a Username attribute in the Component. However, we could show how to add such attribute. But not "Username". Maybe just something like "myAttribute".

bertysentry commented on my-menu-tsps-core/src/main/java/com/bmc/truesight/platform/etl/component/management/types/MyMenuComponent.java:
```diff
@@ -0,0 +1,323 @@
+
+		// Retrieve the User configured in the My Menu Component
+		try {
+			myMenuUser = new SecurityServiceImpl().getUser(super.getServerTenant(), getUsername());
```
I don't think it's necessary, as it's really for authentication matters, which was useful only in Monitoring Studio's case, where we do execute stuff on PATROL Agent "as a user"

bertysentry commented on my-menu-tsps-core/src/test/resources/my-menu-build.properties:
```diff
@@ -0,0 +1 @@
+version=${project.version}
```
This file is not required

bertysentry commented on my-menu-tsps-lib/src/main/java/com/bmc/sas/ngp/service/security/SecurityServiceImpl.java:
```diff
@@ -0,0 +1,22 @@
+
+import com.bmc.sas.domain.user.User;
+
+public class SecurityServiceImpl {
```
Maybe not necessary if you remove the corresponding call

bertysentry commented on my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/Component.java:
```diff
@@ -0,0 +1,171 @@
+
+import java.util.Set;
+
+public class Component {
```
OMG!! I didn't realize you would have to do this! I'm so sorry! And also happy that you manage to do it anyway :-)

NassimBtk replied:
No problem @bertysentry  ;)

bertysentry commented on my-menu-tsps-plugin/src/main/resources/conf/registry.xml:
```diff
@@ -0,0 +1,11 @@
+		<resource startup="false">
+			<className>com.mycompany.mymenu.tsps.controller.MyMenuDeviceProvider</className>
+			<targets>
+				<target authenticate="true" method="GET"><![CDATA[.*/my-menu/.*]]></target>
```
I actually did not expect you to explain how to extend the REST API. But oh well, this is done, great! ;-)

bertysentry commented on my-menu-tsps-web/pom.properties:
```diff
@@ -0,0 +1,4 @@
+#Generated by Apache Maven ${maven.version}
+version=${project.version}
```
Weird: this has been committed with Windows-style CR-LR end-of-lines (it's marked in red by GitHub `^M`). You should probably fix this.

bertysentry commented on my-menu-tsps-web/src/main/webapp/components/my-menu-devices-tsps/my-menu-devices-tsps.html:
```diff
@@ -0,0 +1,39 @@
+<!-- col-sm-12 is kinda mandatory here to avoid unnecessary scrollbars... -->
+<div class="my-menu-tsps col-sm-12">
```
Please add a very simple `<bmc-title>` so future devs see how this works

NassimBtk replied:
Ok let's use the BMC components 
```
<title-section> <title-bar> <title-display>
```

bertysentry commented on my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css:
```diff
@@ -0,0 +1,34 @@
+  width: 30%;
+  max-width: 300px;
+  min-width: 80px;
+  font-family: Verdana, Arial, sans-serif;
```
Do you really have to set *font-family*?

NassimBtk replied:
It is the same :D, but just to show that we can customize things ;)

NassimBtk replied:
Removed

bertysentry commented on my-menu-tsps-web/src/main/webapp/css/my-menu-tsps.css:
```diff
@@ -0,0 +1,34 @@
+  padding-left: 10px;
+  padding-top: 5px;
+  font-size: 3em;
+  font-family: OpenSans;
```
Same here. I'd prefer to encourage future devs to use existing classes and stuff :-)

bertysentry commented on my-menu-tsps-web/src/main/webapp/js/tsps-route.js:
```diff
@@ -0,0 +1,25 @@
+   * By default, myMenuPermissions say we're allowed to access stuff
+   * We ask our REST API what we're allowed to do
+   **/
+  angular
```
This is not needed in this example, please remove.

bertysentry reviewed (changes requested):
Brilliant work, @NassimBtk !
A few minor changes required. 
You will also need to document the installation part, clearly explaining that they need to "Add the Component" to their TSPS (not just install).
Thank you!

NassimBtk committed:
Commit after code review

Updated MyMenuComponent and removed unecessary code

Updated angular template in order to use BMC components

Removed unused css

Update my-menu-devices-tsps.js and removed permission constant

Updated README and added a new section for the Component configuration

bertysentry commented on my-menu-tsps-lib/src/main/java/com/bmc/truesight/api/dto/ComponentAttributeEntity.java:
```diff
@@ -39,4 +103,146 @@ public String getAttributeName() {
 		this.attributeName = attributeName;
 	}
+
+	public String getAttributeValue() {
```
Urgl... sorry again for the pain! ;-)
